src/App.js

```diff
@@ -1,16 +1,29 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [likesCount, setLikesCount] = useState(0);
```

Comment:
That being said, I think there's a huge disadvantage to maintainability by tracking the like counts separately. Namely, it's redundant compared to the messages, since we can calculate the number of likes from the messages themselves.

This is another form of repetition that is usually worth avoiding, as if your code for updating the likes had errors, there is a chance that `likesCount` would fall out of sync with the number of likes in `messages`.

While this lets you avoid an iteration through the messages, this sort of duplication of data is a cause of a lot of bugs, so I think it's worth the extra iteration to make the code "dumber" but less prone to error.